src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot;
+
+import com.aws.greengrass.clientdevices.auth.configuration.RuntimeConfiguration;
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.util.Coerce;
+
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+
+public class CertificateRegistry implements CertificateRepository {
+    private static final String STATUS_KEY = "s";
+
+    private final Topics registryTopics;
+
+    @Inject
+    public CertificateRegistry(RuntimeConfiguration runtimeConfiguration) {
+        this.registryTopics = runtimeConfiguration.getCertRegistry();
+    }
+
+    @Override
+    public Optional<Certificate> getCertificateFromPem(String certificatePem) throws InvalidCertificateException {
+        Certificate cert = Certificate.fromPem(certificatePem);
+        return getCertificateById(cert.getCertificateId());
+    }
+
+    @Override
+    public Certificate createCertificate(String certificatePem) throws InvalidCertificateException {
+        Certificate newCert = Certificate.fromPem(certificatePem);
+        updateCertificate(newCert);
+        return newCert;
+    }
+
+    @Override
+    public void updateCertificate(Certificate certificate) {
+        putCertificate(certificate);
+    }
+
+    @Override
+    public void deleteCertificate(Certificate certificate) {
+        Topics certTopics = registryTopics.lookupTopics(certificate.getCertificateId());
+        certTopics.remove();
+    }
+
+    private void putCertificate(Certificate certificate) {
+        Topics certTopics = registryTopics.lookupTopics(certificate.getCertificateId());
+        long modtime = certificate.getStatusLastUpdated().toEpochMilli();
+        certTopics.lookup(STATUS_KEY).withNewerValue(modtime, certificate.getStatus().toString());
```

Comment:
Why not have this on the RuntimeConfig? - like `RuntimeConfig.updateCertificateStatus(modtime, certificate.getStatus().toString())`. This is not bad, it is just that it feels weird that some methods on the Runtime Config return values and others return topics.